internal/controller/nonadminbackup_controller.go

```diff
@@ -758,11 +764,26 @@ func updateNonAdminBackupVeleroBackupStatus(status *nacv1alpha1.NonAdminBackupSt
 	}
 
 	if reflect.DeepEqual(*status.VeleroBackup.Status, veleroBackup.Status) {
-		return false
+		return false, nil
 	}
 
 	status.VeleroBackup.Status = veleroBackup.Status.DeepCopy()
-	return true
+
+	// adds logsPath and resourceListPath to status if velero backup is completed.
+	if status.VeleroBackup.Status.Phase == velerov1.BackupPhaseCompleted {
```

Comment:
this does not seem to comply with design. Here we only add info if backup succeeds. Should not check for finished state phases?

Reply:
Well the info isn't usable until backup completes. If it is unclear in design will fix.

If backup fails, (I guess partially fail could work), these urls won't work.

Reply:
This sentence made me think that the goal is to add more info for debugging unsuccessful backups/restores
https://github.com/kaovilai/oadp-non-admin/blob/29286ec5cbc3021ba93f5013196d60e2fa98a611/docs/design/Debug_DownloadURL.md?plain=1#L13

maybe adding to `Non Goals` that backup/restore phases that will not contain the info

Reply:
This should work for all the cases where we will have information available in s3 for the corresponding backup.